toolchain/check/handle_literal.cpp

```diff
@@ -58,11 +58,42 @@ auto HandleIntLiteral(Context& context, Parse::IntLiteralId node_id) -> bool {
 }
 
 auto HandleRealLiteral(Context& context, Parse::RealLiteralId node_id) -> bool {
+  // Convert the real literal to an llvm::APFloat and add it to the floats
+  // ValueStore. In the future this would use an arbitrary precision Rational
+  // type.
+  auto real_id =
+      context.tokens().GetRealLiteral(context.parse_tree().node_token(node_id));
+  auto real_value = context.sem_ir().reals().Get(real_id);
+
+  if (real_value.mantissa.getActiveBits() > 64) {
```

Comment:
The constraints that we want to apply here are [described as follows in the design](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/implicit_conversions.md#data-types):

> A floating-point constant can be implicitly converted to any type fM in which that value is between the least representable finite value and the greatest representable finite value (inclusive), and converts to the nearest representable finite value, with ties broken by picking the value for which the mantissa is even.

I think it's fine to apply a different set of checks for now, but if so, please can you add a `TODO` comment to implement the above instead?

Reply:
Added a TODO comment to implement the implicit conversions as documented.

I'm happy to work on it on a follow up PR. I'd just need to read up on floating points first, it's been a while since I've worked on this kind of lowish-level things 😅.